### PR TITLE
[4.0] newsfeeds - nullable datetime fields

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #16788 ....same as #24675 for contacts - I really just copied the work of others (someone has to)

### Summary of Changes
some datetime fields should be nullable as default

- publish_up
- publish_down
- checked_out_time

### Testing Instructions
test com_newsfeeds

### Expected result
newsfeeds should works as before

### CODE REVIEW

File: administrator/components/com_newsfeeds/Table/NewsfeedTable.php [hunk 1]
```diff
@@ -169,16 +177,14 @@ public function store($updateNulls = false)
...
-		// Set publish_up to null date if not set
+		// Set publish_up, publish_down to null if not set
 		if (!$this->publish_up)
 		{
-			$this->publish_up = $this->_db->getNullDate();
+			$this->publish_up = null;
 		}
-
-		// Set publish_down to null date if not set
 		if (!$this->publish_down)
```
Comment (Quy) [inline administrator/components/com_newsfeeds/Table/NewsfeedTable.php:186]:
```suggestion

		if (!$this->publish_down)
```

File: components/com_newsfeeds/Model/CategoryModel.php [hunk 2]
```diff
@@ -163,8 +163,8 @@ protected function getListQuery()
 
 		if ($this->getState('filter.publish_date'))
 		{
-			$query->where('(a.publish_up = ' . $nullDate . ' OR a.publish_up <= ' . $nowDate . ')')
-				->where('(a.publish_down = ' . $nullDate . ' OR a.publish_down >= ' . $nowDate . ')');
+			$query->where('(' . $query->isNullDatetime('a.publish_up') . ' OR a.publish_up <= ' . $db->quote($nowDate) . ')')
+			->where('(' . $query->isNullDatetime('a.publish_down') . ' OR a.publish_down >= ' . $db->quote($nowDate) . ')');
```
Comment (Quy) [inline components/com_newsfeeds/Model/CategoryModel.php]: Revert indentation. See line 144.

File: components/com_newsfeeds/Model/NewsfeedModel.php [hunk 3]
```diff
@@ -116,9 +116,9 @@ public function &getItem($pk = null)
 				if (is_numeric($published))
 				{
 					$query->where('(a.published = ' . (int) $published . ' OR a.published =' . (int) $archived . ')')
-						->where('(a.publish_up = ' . $nullDate . ' OR a.publish_up <= ' . $nowDate . ')')
-						->where('(a.publish_down = ' . $nullDate . ' OR a.publish_down >= ' . $nowDate . ')')
-						->where('(c.published = ' . (int) $published . ' OR c.published =' . (int) $archived . ')');
+					->where('(' . $query->isNullDatetime('a.publish_up') . ' OR a.publish_up <= ' . $db->quote($nowDate) . ')')
```
Comment (Quy) [inline components/com_newsfeeds/Model/NewsfeedModel.php]: Revert indentation.

**Comment by tonypartridge** [comment]
@brianteeman  please review Quy's changes and ill test

**Comment by brianteeman** [comment]
codestyle should be ok now

File: components/com_newsfeeds/Model/CategoryModel.php [hunk 4]
```diff
@@ -182,8 +182,8 @@ protected function getListQuery()
 
 		if ($this->getState('filter.publish_date'))
 		{
-			$query->where('(a.publish_up = ' . $nullDate . ' OR a.publish_up <= ' . $nowDate . ')')
```
Comment (Quy) [inline components/com_newsfeeds/Model/CategoryModel.php:185]: Delete line 179.

File: components/com_newsfeeds/Model/NewsfeedModel.php [hunk 5]
```diff
@@ -116,8 +116,8 @@ public function &getItem($pk = null)
 				if (is_numeric($published))
 				{
 					$query->where('(a.published = ' . (int) $published . ' OR a.published =' . (int) $archived . ')')
-						->where('(a.publish_up = ' . $nullDate . ' OR a.publish_up <= ' . $nowDate . ')')
```
Comment (Quy) [inline components/com_newsfeeds/Model/NewsfeedModel.php:119]: Delete line 109.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 5f0f6ec46783b638afaec9c6d641f810bcacf0fd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25362">issues.joomla.org/tracker/joomla-cms/25362</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 5f0f6ec46783b638afaec9c6d641f810bcacf0fd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25362">issues.joomla.org/tracker/joomla-cms/25362</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25362">issues.joomla.org/tracker/joomla-cms/25362</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by brianteeman** [comment]
thanks

**Pull request merged**